services/web/src/components/RequestBlock/index.js

```diff
@@ -0,0 +1,68 @@
+import { Layout } from 'components';
+import React from 'react';
+import { Button, Header, Dropdown } from 'semantic';
+import CodeBlock from '../CodeBlock';
+import templateCurl from './templates/curl';
+import templateFetch from './templates/fetch';
+
+const options = [
+  {
+    template: templateCurl,
+    value: 'curl',
+    text: 'cURL',
+    language: 'bash',
+  },
+  {
+    template: templateFetch,
+    value: 'fetch',
+    text: 'Fetch',
+    language: 'js',
+  },
+];
+
+export default class FetchBlock extends React.Component {
+  state = {
+    current: 'curl',
+  };
+
+  constructor(props) {
+    super(props);
+    this.templateRef = React.createRef();
+  }
+
+  getData() {
+    return this.props.request;
+  }
+
+  handleOnCopy = () => {
```

Comment:
nitty preferences:

- I prefer `onClick` to `handleClick` as it matches the react props
- I prefer `CopyClick` to just `Copy` as you can clearly delineate the event types being handled

Reply:
so it should be CopyClick ? even though it has nothing todo with the actually click?

Reply:
ah sorry was splitting my reasons up... I prefer `onCopyClick` ... it does have to the click as it's the actual click event itself, no?

Reply:
i will try to loss the handle going forward